File: leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderInteractorDataSource.java
package com.schibstedspain.leku.geocoder;

import android.location.Address;
import com.google.android.gms.maps.model.LatLng;
import io.reactivex.Observable;
import java.util.List;

public interface GeocoderInteractorDataSource {
  Observable<List<Address>> getFromLocationName(String query);

  Observable<List<Address>> getFromLocationName(String query, LatLng lowerLeft, LatLng upperRight);

  Observable<List<Address>> getFromLocation(double latitude, double longitude);
}

File: leku/src/main/java/com/schibstedspain/leku/geocoder/GeocoderInteractorDataSource.kt
package com.schibstedspain.leku.geocoder

import android.location.Address
import com.google.android.gms.maps.model.LatLng
import io.reactivex.Observable

interface GeocoderInteractorDataSource {
    fun getFromLocationName(query: String): Observable<List<Address>>

    fun getFromLocationName(query: String, lowerLeft: LatLng, upperRight: LatLng): Observable<List<Address>>

    fun getFromLocation(latitude: Double, longitude: Double): Observable<List<Address>>
}
